### ISSUE TITLE
Input pristine and value issue

### ISSUE THREAD

**Issue by cristianocca** [formsy/formsy-react#128]
Hello,

I'm trying to update from version 0.x but some changes are giving me problems right now.

When trying to wrap a text field (i.e, from material ui), it's all fine when the "value" prop is set and never changed. But if the parent component changes the value prop, this new value is not set to the "pristineValue" internal state and hence every time I attempt to reset the form the first pristine value is used.

In summary, I have no way to change the default pristine value once set.

Any ideas?

Thanks.

Edit:
I think that changing this to also update the pristine value would do the trick and it shouldn't break anything.
https://github.com/formsy/formsy-react/blob/master/src/Wrapper.js#L94

**Comment by rkuykendall** [comment]
Your source-link corrected from master to a permalink: https://github.com/formsy/formsy-react/blob/1c5cc126f6009167871d9ed3627d9d13a51dd792/src/Wrapper.js#L94

The same line on master today: https://github.com/formsy/formsy-react/blob/838cfbd9c36944ae500eea9e54a23d6569d857e9/src/Wrapper.ts#L184

I don't _think_ `pristineValue` is supposed to change when value changes, or then it would just be the same as value and not the pristine, or original value. @cristianocca could you explain further, or if you have no more interest in this issue, close it?

**Comment by cristianocca** [comment]
I ended up vendoring the library in the project it was being used at and did a bunch of changes so the issue is no longer relevant for us.

El 22 feb. 2020 14:39, Robert Kuykendall <notifications@github.com> escribió:

Your source-link corrected from master to a permalink: https://github.com/formsy/formsy-react/blob/1c5cc126f6009167871d9ed3627d9d13a51dd792/src/Wrapper.js#L94<https://nam01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fformsy%2Fformsy-react%2Fblob%2F1c5cc126f6009167871d9ed3627d9d13a51dd792%2Fsrc%2FWrapper.js%23L94&data=02%7C01%7C%7Cc0cd8dcfe99d48cfcfa308d7b7be2454%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637179899581083897&sdata=7dp1zPaSuKwS2bndK0QFBRUnHtTf8y5pZ9R7DyTDcgM%3D&reserved=0>

The same line on master today: https://github.com/formsy/formsy-react/blob/838cfbd9c36944ae500eea9e54a23d6569d857e9/src/Wrapper.ts#L184<https://nam01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fformsy%2Fformsy-react%2Fblob%2F838cfbd9c36944ae500eea9e54a23d6569d857e9%2Fsrc%2FWrapper.ts%23L184&data=02%7C01%7C%7Cc0cd8dcfe99d48cfcfa308d7b7be2454%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637179899581093891&sdata=v8XplPkqoeKjTyG%2FwZ5yCmBnPIjE3Z6amDUM6k%2BhNjE%3D&reserved=0>

I don't think pristineValue is supposed to change when value changes, or then it would just be the same as value and not the pristine, or original value. @cristianocca<https://nam01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fcristianocca&data=02%7C01%7C%7Cc0cd8dcfe99d48cfcfa308d7b7be2454%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637179899581103886&sdata=IUyWe1FhRtQ%2BkWKvvmD8KEGavO2qovqB76MrhajUq0s%3D&reserved=0> could you explain further, or if you have no more interest in this issue, close it?

**Comment by rkuykendall** [comment]
@cristianocca Thank you for your response. If you're interested in up-streaming your changes and re-entering the open-source organization, I think you'll find the project in much better shape than originally. If not, I wish your project all the best!
